### PR TITLE
File browser reshuffle

### PR DESCRIPTION
**Pull request by Commodore64user**
From the makers of the wildly successful Sleep screen rework #11549 and the yet to be merged status bar makeover #11678, We present to you the 'new kid in town', the new file browser (file cabinet) menu.

There has been a far less aggressive approach here, most of the work pertains the reduction of "view modes" and rewording of settings. Some logic was added for the functioning of the simpler view modes but besides that no changes in functionality were added or removed.

I am conflicted about the use of the word "mosaic" here, as what we get is a standard grid view that has nothing to do with actual mosaics, I would love to hear what you think, should it go, should it stay?

the following screenshots were captured on a non-touch kindle 4, so some options might not appear there. Anyway without any further ado, 

Here it is the menu upon opening:

after going to the `Library view`:

Inside `mosaic and list view properties`:

and finally, inside the `file management` submenu:

> [!IMPORTANT]
> I would like to have the classic view properties greyed out when 'classic mode' NOT selected but the usual `enable function` does not work in that case, any help there would be greatly appreciated.

Last but not least, thanks to @Frenzie for his help, patience and support.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/11781)

### CODE REVIEW

File: frontend/apps/filemanager/filemanagermenu.lua [hunk 1]
```diff
@@ -885,7 +885,7 @@ function FileManagerMenu:getStartWithMenuTable()
     local start_withs = {
         { _("file browser"), "filemanager" },
         { _("history"), "history" },
-        { _("favorites"), "favorites" },
+        { _("collections"), "favorites" },
```
Comment (hius07) [inline frontend/apps/filemanager/filemanagermenu.lua:888]: We actually start with Favorites, not with collection list.

File: plugins/coverbrowser.koplugin/main.lua [hunk 2]
```diff
@@ -126,47 +131,186 @@ function CoverBrowser:addToMainMenu(menu_items)
...
         end,
     })
     table.insert(sub_item_table, {
-        text = _("History display mode"),
+        text = _("Replace book title with file name"),
         enabled_func = function()
-            return not BookInfoManager:getSetting("unified_display_mode")
+            return filemanager_display_mode == "list_image_meta"
+                or filemanager_display_mode == "list_image_filename"
+                or filemanager_display_mode == "list_only_meta"
```
Comment (hius07) [inline plugins/coverbrowser.koplugin/main.lua:162]: Maybe you can use something like
https://github.com/koreader/koreader/blob/526a1fb7276468137045d6227c8f62b58c75daa3/plugins/coverbrowser.koplugin/main.lua#L513
to shorten this and similar checks.

File: frontend/apps/filemanager/filemanagermenu.lua [hunk 3]
```diff
@@ -165,17 +165,32 @@ function FileManagerMenu:setUpdateItemTable()
...
                 sub_item_table = {
                     {
                         text_func = function()
-                            return T(_("Items per page: %1"),
+                            return T(_("Book entries per page: %1"),
                                 G_reader_settings:readSetting("items_per_page") or FileChooser.items_per_page_default)
                         end,
-                        help_text = _([[This sets the number of items per page in:
+                        help_text = _([[This sets the number of entries per page in:
 - File browser, history and favorites in 'classic' display mode
```
Comment (hius07) [inline frontend/apps/filemanager/filemanagermenu.lua:194]: "... and collections ..."

File: frontend/apps/filemanager/filemanagermenu.lua [hunk 4]
```diff
@@ -295,7 +310,7 @@ function FileManagerMenu:setUpdateItemTable()
                         separator = true,
                     },
                     {
-                        text = _("Clear history of deleted files"),
+                        text = _("Clear deleted files from history"),
                         callback = function()
                             UIManager:show(ConfirmBox:new{
                                 text = _("Clear history of deleted files?"),
```
Comment (hius07) [inline frontend/apps/filemanager/filemanagermenu.lua:316]: May be changed too.

**Comment by poire-z** [comment]
This PR conflicts and does not apply on a current nightly.
image

It's been a while since you started your work, and you haven't updated it for the recent changes to coverbrowser.main.lua, the top 2 commits in:
https://github.com/koreader/koreader/commits/master/plugins/coverbrowser.koplugin/main.lua

Dunno how this is solvable via GitHub web (dunno if you're using that or a real git command line), @hius07 may know what's the best way to resolve it.
Had to manually edit that file on my device, starting from yours, and re-applying these last 2 commits changes (2 "favorites" <> "collections, one `local fc = ...` to add, 6 to remove, 11 self:refreshFileManagerInstance() to replace with fc:updateItems(1, true)), not sure I got it all right, but if you resolve them manually, here's my final file if you need to compare: [fixed_manually_main.lua.txt](https://github.com/koreader/koreader/files/15266204/fixed_manually_main.lua.txt)

**Comment by Commodore64user** [comment]
okay @poire-z I believe I resolved all the conflicts...?

**Comment by benoit-pierre** [comment]
No, you need to rebase, you're still 47 commits behind master.

**Pull request closed without merging**